dff/script/core/context.py

```diff
@@ -144,38 +140,37 @@ def cast(cls, ctx: Optional[Union["Context", dict, str]] = None, *args, **kwargs
         if not ctx:
             ctx = Context(*args, **kwargs)
         elif isinstance(ctx, dict):
-            ctx = Context.parse_obj(ctx)
+            ctx = Context.model_validate(ctx)
         elif isinstance(ctx, str):
-            ctx = Context.parse_raw(ctx)
+            ctx = Context.model_validate_json(ctx)
         elif not issubclass(type(ctx), Context):
             raise ValueError(
                 f"context expected as sub class of Context class or object of dict/str(json) type, but got {ctx}"
             )
         return ctx
 
-    @validate_arguments
```

Comment:
We can but that is not how we are supposed to work with pydantic.

Reply:
In my view, this strategy is better, since it does not modify the `Context` model, while using `validate_call` decorators leads to the `Message` class being referenced in the Context class body which causes the pickling problems